client/src/utils/components/ssh/ssh.tsx

```diff
@@ -0,0 +1,223 @@
+  const projectToShowGitUrl = cleanGitUrl(displayModal.gitUrl);
+  if (useLocalMigration && projectCurrentGitUrl !== projectToShowGitUrl && !localMigration.fetching) {
+    setLocalMigration({ data: { gitUrl: projectToShowGitUrl }, fetched: false, fetching: true });
+    client.checkMigration(projectToShowGitUrl).then((resp: any) => {
+      if (resp?.result)
+        setLocalMigration({ fetched: true, fetching: false, data: { ...resp.result, gitUrl: projectToShowGitUrl } });
+      else if (resp?.error)
+        setLocalMigration({ fetched: true, fetching: false, data: { ...resp.error, gitUrl: projectToShowGitUrl } });
+    });
+  }
+
+  const loading = notebooksSupport.isLoading ||
+    (!useLocalMigration && !projectMigrationData?.core_renku_version) ||
+    (useLocalMigration && localMigration.fetching) ? true : false;
+  const errorCore = (!useLocalMigration && projectMigrationData?.check_error?.code) ||
+    (useLocalMigration && projectMigrationData.code) ? true : false;
+  const sshCoreSupport = projectMigrationData.template_status?.ssh_supported ?? false;
+  const sshNotebooksSupport = notebooksSupport.data?.sshEnabled ?? false;
+
+  const toggleModal = () => dispatch(toggleSshModal());
+  const closeModal = () => dispatch(hideSshModal());
+
+  let modalBody: React.ReactNode;
+  // ? mind that this first case should not happen since modal shouldn't even open for dpeloyment without SSH support.
+  if (!sshNotebooksSupport) {
+    modalBody = (
+      <>
+        <p>This RenkuLab deployment doesn{"'"}t support SSH.</p>
+        <p>Please contact your local administrator for further details.</p>
+      </>
+    );
+  }
+  else if (errorCore) {
+    modalBody = (<>
+      <p>
+        An error occurred while verifying SSH support on your project.
+        Please check the following message for further details.
+      </p>
+      <p>
+        Please refer to{" "}
+        <ExternalDocsLink url={Docs.rtdHowToGuide("renkulab/ssh-into-sessions.html")}
+          title="our documentation" /> to get further information.
+      </p>
+      <GeneralErrorMessage
+        error_while="checking"
+        error_what="project template"
+        error_reason={
+          (projectMigrationData.check_error?.userMessage || projectMigrationData.userMessage) ?? ""
+        }
+      />
+    </>);
+  }
+  else if (loading) {
+    modalBody = <Loader />;
+  }
+  else if (!sshCoreSupport) {
+    const updateUrl = Url.get(Url.pages.project.overview.status, { namespace: "", path: displayModal.projectPath });
+    modalBody = (
+      <>
+        <p>
+          <img src={rkIconSshCross} className="rk-icon rk-icon-md filter-dark me-2" />
+          Your project needs to be updated to support SSH.
+        </p>
+        <p>
+          To update your project, go to{" "}
+          <Link to={updateUrl} onClick={() => closeModal()}>Overview Status</Link>
+          {" "}then come back to this window to view the commands to connect remotely to sessions.
```

Comment:
I think "come back to this window" is a bit misleading, since the window disappears when you go to the status page. Might better to say:

```suggestion
          {" "}start an ssh session again to view the commands to connect remotely to sessions.
```

Reply:
I further rephrased it based on both yours and other offline comments.
Please let me know if it works well